lib/chat_api/users.ex

```diff
@@ -8,32 +8,40 @@ defmodule ChatApi.Users do
 
   alias ChatApi.Users.{User, UserProfile, UserSettings}
 
+  @spec find_user_by_email(binary()) :: User.t() | nil
   def find_user_by_email(email) do
     User |> where(email: ^email) |> Repo.one()
   end
 
+  @spec find_user_by_email(binary() | nil, binary()) :: User.t() | nil
   def find_user_by_email(nil, _account_id), do: nil
 
   def find_user_by_email(email, account_id) do
     User |> where(account_id: ^account_id, email: ^email) |> Repo.one()
   end
 
+  @spec find_by_id!(binary()) :: User.t()
```

Comment:
apologies -- the `users` table is the only one that actually uses integers for the ID 😅

Reply:
😭